python/pyspark/pandas/spark/functions.py

```diff
@@ -80,9 +80,22 @@ def repeat(col: Column, n: Union[int, Column]) -> Column:
     """
     Repeats a string column n times, and returns it as a new string column.
     """
-    sc = SparkContext._active_spark_context
-    n = _to_java_column(n) if isinstance(n, Column) else _create_column_from_literal(n)
-    return _call_udf(sc, "repeat", _to_java_column(col), n)
+    if is_remote():
+        from pyspark.sql.connect.functions import lit, call_udf
+
+        if isinstance(n, int):
+            n = lit(n)  # type: ignore[assignment]
+
+        return call_udf(  # type: ignore[return-value]
+            "repeat",
+            col,  # type: ignore[arg-type]
+            n,  # type: ignore[arg-type]
+        )
+
+    else:
+        sc = SparkContext._active_spark_context
+        n = _to_java_column(n) if isinstance(n, Column) else _create_column_from_literal(n)
+        return _call_udf(sc, "repeat", _to_java_column(col), n)
```

Comment:
actually I think we can just call `call_udf` instead without if-else.

Reply:
let me take another look